src/Xamarin.CodeAnalysis/XI0004.cs

```diff
@@ -0,0 +1,177 @@
+				var location = codeBlockContext.OwningSymbol.Locations.FirstOrDefault ();
+				if (location == null)
+					return;
+				var diagnostic = Diagnostic.Create (Rule, location, method.Name);
+				codeBlockContext.ReportDiagnostic (diagnostic);
+			});
+		}
+
+	}
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(RequiresSuperAnalyzerCodeFixProvider))]
+    [Shared]
+    public class RequiresSuperAnalyzerCodeFixProvider : CodeFixProvider
+    {
+        const string Title = "Add base method call";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(XI0004.DiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            // Find the method declaration identified by the diagnostic.
+            var methodDeclaration =
+                root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf()
+                .OfType<MethodDeclarationSyntax>().First();
+
+            // Register a code action that will invoke the fix.
+            context.RegisterCodeFix(CodeAction.Create(Title, c => FixSuperAsync(context.Document, methodDeclaration, c), equivalenceKey: Title), diagnostic);
+        }
+
+        async Task<Document> FixSuperAsync(Document document,
+          MethodDeclarationSyntax methodDeclaration,
+          CancellationToken cancellationToken)
+        {
+            //Find the node to fix.
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var memberSymbol = semanticModel.GetDeclaredSymbol(methodDeclaration) as IMethodSymbol;
+            if (memberSymbol == null)
+                return document;
+
+            //Generate the new base.[methodName]([arguments]) block. If the method needs to return the base call,
+            //it generates return base.[methodName]([arguments])
+            // TODO: this might be hacky - (.Add(SyntaxFactory.Tab) adds an extra tab so that it's a tab (or SyntaxFactory.Tab) further than the leading
+            //      trivia of the method body block, but we should consider investigating if this behaves as expected when the IDE is set to spaces vs
+            //      tabs / different numbers of spaces per tab
+            var returnString = memberSymbol.ReturnsVoid ? "" : "return ";
+            var parametersString = string.Join(",", memberSymbol.Parameters.Select((p) => p.Name).ToArray()); //creates a string of the method's passed-in arguments separated by commas
```

Comment:
Still no need to ToArray

Reply:
ah sorry - I removed ToArray() and thought I committed it but think I must have forgotten to save, I've attached committed it now 😂